app/routes/goal_routes.py

```diff
@@ -1 +1,110 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, abort, make_response
+from app.models.goal import Goal
+from app.models.task import Task
+from app.db import db
+from app.routes.route_utilities import validate_model
+
+bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    title = request_body["title"]
+
+    new_goal = Goal(title=title)
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+    
+
+@bp.get("")
+def get_all_goals():
+    query = db.select(Goal).order_by(Goal.id)
+    goals = db.session.scalars(query)
+
+    return [goal.to_dict() for goal in goals]
+    
+
+@bp.get("/<goal_id>")
+def get_single_goal(goal_id):
+
+    goal = validate_model(Goal, goal_id)
+    return {"goal": goal.to_dict()}
+
+
+@bp.put("/<goal_id>")
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
```

Comment:
Isn't this code similar to our other `.put` method? How could we use the `hasattr` and `setattr` functions that Python provides in order to make DRY up our routes?